src/passes/rejectUnsupportedFeatures.ts

```diff
@@ -86,6 +89,13 @@ export class RejectUnsupportedFeatures extends ASTMapper {
     this.visitExpression(node, ast);
   }
 
+  visitParameterList(node: ParameterList, ast: AST): void {
+    // any of node.vParameters has indexed flag true then throw error
+    if (node.vParameters.some((param) => param.indexed)) {
+      throw new WillNotSupportError(`Indexed parameters are not supported`);
+    }
+  }
```

Comment:
Either this function needs to recurse, or the check should go in visitVariableDeclaration